### ISSUE TITLE
Lap: automatically infer lap

### ISSUE THREAD

**Issue by guancio** [Runalyze/Runalyze#1687]
Automatically infer lap from GPS coordinates.
Possible interface:
1) User selects an activity
2) User opens the edit window
we assume that for the current activity, no lap has been manually added
3) User clicks "infer laps"
4) Runanalyze scans the GPS coordinates and identifies points that are periodically repeated in the track. It selects on of them and split the laps accordingly.
5) The user is redirected to the updated edit page, where the laps have been added

**Comment by mipapo** [comment]
+1 Great idea

**Comment by laufhannes** [comment]
What is your use case for this?

To find repeating coordinates isn't that easy, as the coordinates themselves will probably never repeat - only similar ones. And if I'm on a track or any round course, every point is periodically repeated?

**Comment by mipapo** [comment]
@laufhannes But with our new geohashes we could easily extend the search area for similiar ones. 
And a use case is e.g. being on a track. 
Maybe just the starting point should be used to find repeating points

**Comment by guancio** [comment]
long term use cases include match of segments (where coordinates are similar).
I agree that the coordinates never repeat, but we can use a threshold to identify repetitions e.g.
- let say we have two samples on the track data i and j
- distance(i.point, j.point) < threshold1 and
- track_distance_from_to(i,j) > threshold2
then i and j can be considered a repetition.
I agree that if you run on a track you will have every point in the track repeated periodically. We can
1) just use the first point the is repeated
2) ask the user (I do not know how) where is the track start/end
The latter option can be interesting together with routes. If you run several times on the same track, then you should define the start of the track once, and this can be used by all activities on such track.

**Comment by laufhannes** [comment]
It's not that I don't see the use case - I'm just curious if an automated calculation is the right way to solve the issue. I'd guess it's much easier to pick single points on the map to select them as starting points for new laps.

Using a threshold is clear, but that threshold will be valid for every direction (not only left/right but ahead/behind as well).

**Comment by guancio** [comment]
I agree. About automation, I'm mainly interested in automating the identification of laps, not where a lap should start end. We can just ask the user to identify (using a map?) where the laps start, then we can compute automatically the splits.
